src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java

```diff
@@ -167,7 +169,8 @@ private SupplierFactory() {}
 
     private static <T> ThrowingSupplier<T> mapConstructorParameters(Constructor<T> constructor, Supplier<?>[] suppliers) {
         switch (constructor.getParameterCount()) {
-            case  0: return () -> constructor.newInstance();
+            //that is way to fugly but probably no way to not have it
+            case  0: return constructor::newInstance;
```

Comment:
I'd prefer to keep all this code consistent, this is why I didn't use method reference here.